ckanext/stadtzhtheme/controller.py

```diff
@@ -0,0 +1,171 @@
+from urllib import urlencode
+import logging
+
+from ckan.plugins import toolkit as tk
+import ckan.model as model
+import ckan.logic as logic
+import ckan.lib.plugins
+from ckan.common import c, config, _, request, OrderedDict
+import ckan.lib.helpers as h
+import ckan.lib.search as search
+import ckan.lib.base as base
+from six import string_types
+
+
+import ckan.controllers.group as group
+
+lookup_group_controller = ckan.lib.plugins.lookup_group_controller
+log = logging.getLogger(__name__)
+get_action = logic.get_action
+NotFound = logic.NotFound
+NotAuthorized = logic.NotAuthorized
+ValidationError = logic.ValidationError
+abort = tk.abort
+render = base.render
+
+
+class OgdzhGroupSearchController(group.GroupController):
+    """
+    This controller is copied over from the groups.py controller
+    of CKAN version 2.8. As soon as the update to 2.8 has been
+    done this patch is not necessary any more.
+    This overwrite forces CKAN to use the 'fq' parameter
+    instead of the 'q' parameter for filterig by groups
+    in the solr search query.
+    """
+
+    def _read(self, id, limit, group_type): # noqa
+        ''' This is common code used by both read and bulk_process'''
+        context = {'model': model, 'session': model.Session,
+                   'user': c.user,
+                   'schema': self._db_to_form_schema(group_type=group_type),
+                   'for_view': True, 'extras_as_string': True}
+
+        q = c.q = request.params.get('q', '')
+        # Search within group
+        if c.group_dict.get('is_organization'):
+            fq = 'owner_org:"%s"' % c.group_dict.get('id')
```

Comment:
Can you also add a comment here to make clear which parts of this function were changed?
It would also be helpful to link the PR or commit from CKAN itself in a TODO comment here, so we can quickly check in the future, if we can remove this. 
Similar to how it is noted here https://github.com/opendatazurich/ckanext-stadtzh-theme/blob/1da2479e33f12b83daa2f477db35af4f03fbff8d/ckanext/stadtzhtheme/plugin.py#L196 but with a TODO.
What do you think?